bot/exts/holidays/pride/pride_facts.py

```diff
@@ -28,65 +27,45 @@ def __init__(self, bot: Bot):
-        current_year_facts = FACTS.get(str(now.year), [])[:now.day]
-        previous_facts = current_year_facts + [x for y in previous_years_facts for x in y]
+    async def send_select_fact(self, target: discord.abc.Messageable, day_num: int) -> None:
+        """Provides the fact for the specified day."""
         try:
-            await ctx.send(embed=self.make_embed(random.choice(previous_facts)))
+            await target.send(embed=self.get_fact_embed(day_num))
         except IndexError:
-            await ctx.send("No facts available")
-
-    async def send_select_fact(self, target: discord.abc.Messageable, _date: str | datetime) -> None:
-        """Provides the fact for the specified day, if the day is today, or is in the past."""
-        now = datetime.now(tz=UTC)
-        if isinstance(_date, str):
-            try:
-                date = dateutil.parser.parse(_date, dayfirst=False, yearfirst=False, fuzzy=True)
-            except (ValueError, OverflowError) as err:
-                await target.send(f"Error parsing date: {err}")
-                return
-        else:
-            date = _date
-        if date.year < now.year or (date.year == now.year and date.day <= now.day):
-            try:
-                await target.send(embed=self.make_embed(FACTS[str(date.year)][date.day - 1]))
-            except KeyError:
-                await target.send(f"The year {date.year} is not yet supported")
-                return
-            except IndexError:
-                await target.send(f"Day {date.day} of {date.year} is not yet support")
-                return
-        else:
-            await target.send("The fact for the selected day is not yet available.")
+            await target.send(f"Day {day_num} is not supported")
+            return
 
     @commands.command(name="pridefact", aliases=("pridefacts",))
-    async def pridefact(self, ctx: commands.Context, option: str = None) -> None:
+    async def pridefact(self, ctx: commands.Context, option: int | str = None) -> None:
         """
         Sends a message with a pride fact of the day.
 
-        If "random" is given as an argument, a random previous fact will be provided.
-
-        If a date is given as an argument, and the date is in the past, the fact from that day
-        will be provided.
+        "option" is an optional setting, which has two has two accepted values:
+          - "random": a random previous fact will be provided.
+          - If a option is a number (1-30), the fact for that given day of June is returned.
         """
         if not option:
-            await self.send_select_fact(ctx, datetime.now(tz=UTC))
+            await self.send_select_fact(ctx, datetime.now(tz=UTC).day)
         elif option.lower().startswith("rand"):
-            await self.send_random_fact(ctx)
-        else:
+            await ctx.send(embed=self.get_fact_embed())
+        elif isinstance(option, int):
```

Comment:
This needs to be before the previous check, otherwise `.lower()` will be called on the integer and cause an error.

Reply:
Good point, fixed in [forcepush](https://github.com/python-discord/sir-lancebot/compare/b0dc052e9a426511296f83b9f12871fbd4bd4ac4..ab551fa4efbf1001b4e5a14326577dc7c4ee962e)